tracer/src/Datadog.Trace/Debugger/Instrumentation/AsyncMethodDebuggerInvoker.cs

```diff
@@ -0,0 +1,371 @@
+            asyncState.HasLocalsOrReturnValue = true;
+            asyncState.HasArguments = false;
+        }
+
+        /// <summary>
+        /// Logs the given <paramref name="local"/> ByRef.
+        /// </summary>
+        /// <typeparam name="TLocal">Type of local.</typeparam>
+        /// <param name="local">The local to be logged.</param>
+        /// <param name="index">index of given argument.</param>
+        /// <param name="asyncState">Debugger asyncState</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void LogLocal<TLocal>(ref TLocal local, int index, ref AsyncMethodDebuggerState asyncState)
+        {
+            if (!asyncState.IsActive)
+            {
+                return;
+            }
+
+            var localNamesFromPdb = asyncState.MethodMetadataInfo.LocalVariableNames;
+            if (!MethodDebuggerInvoker.TryGetLocalName(index, localNamesFromPdb, out var localName))
+            {
+                return;
+            }
+
+            LogLocal(ref local, typeof(TLocal), localName, index, ref asyncState);
+        }
+
+        /// <summary>
+        /// End Method with Return value invoker - the MoveNext method always return void but here we send the kick-of method return value
+        /// This method is called from either (1) the outer-most catch clause when the async method threw exception
+        /// or (2) when the async method has logically ended.
+        /// In this phase we have the correct async context in hand because we already set it in the BeginMethod.
+        /// </summary>
+        /// <typeparam name="TTarget">Target object of the method. Note that it could be typeof(object) and not a concrete type</typeparam>
+        /// <param name="instance">Instance value</param>
+        /// <param name="exception">Exception value</param>
+        /// <param name="asyncState">Debugger asyncState</param>
+        /// <returns>LiveDebugger return structure</returns>
+        public static DebuggerReturn EndMethod_StartMarker<TTarget>(TTarget instance, Exception exception, ref AsyncMethodDebuggerState asyncState)
+        {
+            if (!asyncState.IsActive)
+            {
+                return DebuggerReturn.GetDefault();
+            }
+
+            asyncState.SnapshotCreator.StartReturn();
+            asyncState.SnapshotCreator.CaptureInstance(asyncState.KickoffInvocationTarget, asyncState.KickoffInvocationTargetType);
+            if (exception != null)
+            {
+                asyncState.SnapshotCreator.CaptureException(exception);
+            }
+
+            EndMethodLogLocals(ref asyncState);
+            BeginEndMethodLogArgs(ref asyncState);
+
+            return DebuggerReturn.GetDefault();
+        }
+
+        /// <summary>
+        /// End Method with Return value invoker - the MoveNext method always return void but here we send the kick-of method return value
```

Comment:
```suggestion
        /// End Method with Return value invoker - the MoveNext method always returns void but here we send the kick-off method's return value
```

Reply:
Fixed in 5a103fbe57cedbc9e13568d4372c82a921c46e26